### PR TITLE
fix: "geo" params now use relaxed parameter  BREAKING CHANGE: coordinates of "0" are no longer valid

### PR DESCRIPTION
**Pull request by krackerman**
…laxed disabled

### CODE REVIEW

File: src/params/geo-point.js [hunk 1]
```diff
@@ -20,6 +24,9 @@ class GeoPoint extends NumberList {
       ) {
         valid = false;
       }
+      if (this.relaxed === false && (valueParsed[0] === 0 || valueParsed[1] === 0)) {
```
Comment (simlu) [inline src/params/geo-point.js]: fix

Comment (simlu) [inline src/params/geo-point.js]: fix, and explain to me why it needs fixing :D

File: src/params/geo-shape.js [hunk 2]
```diff
@@ -15,6 +16,8 @@ class GeoShape extends Json {
     this.clockwise = clockwise;
     this.type = 'array';
     this.items = { type: 'array', items: { type: 'number' } };
+    assert(relaxed === undefined || typeof relaxed === 'boolean');
```
Comment (simlu) [inline src/params/geo-shape.js]: not sure about assert placement on all of these

Comment (krackerman) [reply src/params/geo-shape.js]: Moved them all to be right under constructor.

File: src/params/geo-point.js (hunk 1 above)
Comment (krackerman) [reply src/params/geo-point.js]: if valid was false from previous check it is redundant to do this check.

Quick thought. Should we refactor the validate in "geo-point.js" and "geo-rect.js" to be more like "str-list.js"?

Comment (simlu) [reply src/params/geo-point.js]: It's actually much worse than just inefficient. Know why?

Yeah, let's do that

File: src/params/geo-point.js [hunk 3]
```diff
@@ -1,25 +1,32 @@
+const assert = require('assert');
+const get = require('lodash.get');
 const NumberList = require('./number-list');
 
 class GeoPoint extends NumberList {
-  constructor(...args) {
-    super(...args);
+  constructor(name, position, opts = {}) {
+    assert(opts.relaxed === undefined || typeof opts.relaxed === 'boolean');
```
Comment (simlu) [inline src/params/geo-point.js]: below super

File: src/params/geo-point.js [hunk 4]
```diff
@@ -1,25 +1,32 @@
...
-        || valueParsed[1] < -90
-        || valueParsed[1] > 90
-      ) {
-        valid = false;
-      }
+    let valueParsed = value;
+    if (valid && this.stringInput) {
+      valueParsed = JSON.parse(value);
+    }
+    if (valid && (valueParsed.length !== 2
```
Comment (simlu) [inline src/params/geo-point.js]: fix breaks

File: src/params/geo-point.js [hunk 5]
```diff
@@ -1,25 +1,32 @@
...
+      || valueParsed[0] < -180
+      || valueParsed[0] > 180
+      || valueParsed[1] < -90
+      || valueParsed[1] > 90)
+    ) {
+      valid = false;
+    }
+    if (valid && this.relaxed === false && (valueParsed[0] === 0 || valueParsed[1] === 0)) {
```
Comment (simlu) [inline src/params/geo-point.js]: fix

File: test/params/geo-point.spec.js [hunk 6]
```diff
@@ -2,8 +2,9 @@ const expect = require('chai').expect;
 const api = require('../../src/index').Api();
 
 describe('Testing GeoPoint Parameter', () => {
-  const queryParam = api.GeoPoint('geoPoint', 'query');
-  const jsonParam = api.GeoPoint('geoPoint', 'json');
+  const queryParam = api.GeoPoint('geoPoint', 'query', { relaxed: true });
```
Comment (simlu) [inline test/params/geo-point.spec.js]: don't change these

**Pull request merged**